File: Dynamic Programming/cpp/55.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int reach = nums.size()-1;
        for(int i=reach;i>=0;i--){
            if(nums[i]+i>=reach){
                reach=i;
            }
        }
        if(reach==0){
            return true;
        }
        return false;
    }
};

File: Dynamic Programming/java/55.java
class Solution {
    public boolean canJump(int[] nums) {
        int reach = nums.length - 1;
        for (int i = reach; i >= 0; i--) {
            if (nums[i] + i >= reach) {
                reach = i;
            }
        }
        return reach == 0;
    }
}